### PR TITLE
PACT-870: updating legal status request-reply to match revised schema

### PR DESCRIPTION
**Pull request by rwalpole**
Please note that the functional changes in this PR are all [in this commit](https://github.com/nationalarchives/ctd-omega-editorial-frontend/pull/115/commits/eae3bd4e9e81d13fb3448a55343efb904b2077d7). The other commits are renaming an re-organising things.

### CODE REVIEW

File: app/uk/gov/nationalarchives/omega/editorial/connectors/messages/ReplyMessage.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.editorial.connectors.messages
```
Comment (userjd) [inline app/uk/gov/nationalarchives/omega/editorial/connectors/messages/ReplyMessage.scala:45]: duplicate package

File: app/uk/gov/nationalarchives/omega/editorial/connectors/RequestReplyHandler.scala [hunk 2]
```diff
@@ -25,6 +25,8 @@ import cats.effect.IO
 import cats.effect.std.Queue
 import org.typelevel.log4cats.Logger
 import uk.gov.nationalarchives.omega.editorial.connectors.JmsRequestReplyClient.ReplyMessageHandler
+import uk.gov.nationalarchives.omega.editorial.connectors.messages.RequestMessage
+import uk.gov.nationalarchives.omega.editorial.connectors.messages.uk.gov.nationalarchives.omega.editorial.connectors.messages.ReplyMessage
```
Comment (userjd) [inline app/uk/gov/nationalarchives/omega/editorial/connectors/RequestReplyHandler.scala:29]: duplicate package name

File: app/uk/gov/nationalarchives/omega/editorial/services/jms/StubServer.scala [hunk 3]
```diff
@@ -81,8 +82,25 @@ class StubServer @Inject() (responseBuilder: ResponseBuilder) {
...
-    } yield responseMessage
+      replyMessage     <- messageFactory.makeTextMessage(responseText)
+      _ = replyMessage.setJMSCorrelationId(requestMessageId)
+      _ = replyMessage.setStringProperty(MessageProperties.OMGApplicationID, "PACS001")
+      _ = replyMessage.setStringProperty(
+            MessageProperties.OMGMessageTypeID,
+            getReplyMessageType(jmsMessage.getStringProperty(MessageProperties.OMGMessageTypeID))
+          )
+      _ = replyMessage.setStringProperty(MessageProperties.OMGMessageFormat, "application/json")
+      _ = replyMessage.setStringProperty(MessageProperties.OMGReplyAddress, "PACS001.request")
```
Comment (userjd) [inline app/uk/gov/nationalarchives/omega/editorial/services/jms/StubServer.scala:93]: Is the reply queue `omega-editorial-web-application-instance-1`

File: app/uk/gov/nationalarchives/omega/editorial/services/jms/StubServer.scala [hunk 4]
```diff
@@ -81,8 +82,25 @@ class StubServer @Inject() (responseBuilder: ResponseBuilder) {
...
+      _ = replyMessage.setStringProperty(MessageProperties.OMGToken, "AbCdEf123456")
+    } yield replyMessage
+
+  private val messageTypeMap = Map[String, String]("OSLISALS001" -> "ODLISALS001")
+  private def getReplyMessageType(maybeMessageType: Option[String]): String = {
```
Comment (userjd) [inline app/uk/gov/nationalarchives/omega/editorial/services/jms/StubServer.scala:98]: why does the reply `MessageTypeId` change here?

File: app/uk/gov/nationalarchives/omega/editorial/connectors/messages/ReplyMessage.scala (hunk 1 above)
Comment (rwalpole) [reply app/uk/gov/nationalarchives/omega/editorial/connectors/messages/ReplyMessage.scala]: Removed.

File: app/uk/gov/nationalarchives/omega/editorial/connectors/RequestReplyHandler.scala (hunk 2 above)
Comment (rwalpole) [reply app/uk/gov/nationalarchives/omega/editorial/connectors/RequestReplyHandler.scala]: Removed.

File: app/uk/gov/nationalarchives/omega/editorial/services/jms/StubServer.scala (hunk 3 above)
Comment (rwalpole) [reply app/uk/gov/nationalarchives/omega/editorial/services/jms/StubServer.scala]: No. In the API schema @adamretter has specified that all messages must have an `OMGReplyAddress` property set which contains the queue name for message replies, and that matches [a regex shown here](https://github.com/nationalarchives/ctd-omega-services/blob/734fef1387aa4145a27636ea7fdafbc3ae76fe9b/schema/omega-services.async.api.yaml#L380). This regex requires a dot separator but it seems that neither SQS nor ElasticMQ support dots in the queue name. We will need to discuss this further with Adam.

File: app/uk/gov/nationalarchives/omega/editorial/services/jms/StubServer.scala (hunk 4 above)
Comment (rwalpole) [reply app/uk/gov/nationalarchives/omega/editorial/services/jms/StubServer.scala:98]: There is a different `MessageTypeId` for the request and reply so in this case `OSLISALS001` is for the request and `ODLISALS001` is for the reply. This information is used to indicate how to unmarshall the reply payload. We are not doing anything with this information at the moment but I wanted it to be as realistic as possible for when need this.

**Comment by rwalpole** [comment]
@userjd I created [PACT-941](https://national-archives.atlassian.net/browse/PACT-941) to add scalafmt to the GitHub Actions for this project.

File: test/support/BaseSpec.scala [hunk 5]
```diff
@@ -73,7 +72,7 @@ class BaseSpec
   )
   val validSessionToken: String = Session.generateToken("1234")
   val invalidSessionToken: String = Session.generateToken("invalid-user")
-  implicit val apiConnectorMonitoring: ApiConnectorMonitoring = TestApiConnector
+  implicit val apiConnectorMonitoring: MessagingServiceMonitoring = TestMessagingService
   lazy implicit val testTimeProvider: TimeProvider = () => LocalDateTime.of(2023, Month.FEBRUARY, 28, 1, 1, 1)
```
Comment (userjd) [inline test/support/BaseSpec.scala:76]: could the variable be renamed to `messagingServiceMonitoring` for consistency

File: app/uk/gov/nationalarchives/omega/editorial/services/jms/StubServer.scala (hunk 3 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/services/jms/StubServer.scala]: OK

File: test/support/BaseSpec.scala (hunk 5 above)
Comment (rwalpole) [reply test/support/BaseSpec.scala:76]: Renamed.

**Pull request merged**